lib/galaxy/tool_util/linters/datatypes.py

```diff
@@ -5,20 +5,22 @@
 )
 
 # from galaxy import config
+from galaxy.tool_util import linters
 from galaxy.tool_util.lint import Linter
 from galaxy.util import (
     listify,
     parse_xml,
 )
+from galaxy.util.resources import resource_path
 
 if TYPE_CHECKING:
     from galaxy.tool_util.lint import LintContext
     from galaxy.tool_util.parser import ToolSource
 
-DATATYPES_CONF = os.path.join(os.path.dirname(__file__), "datatypes_conf.xml.sample")
+DATATYPES_CONF = resource_path(linters, "datatypes_conf.xml.sample")
 
 
-def _parse_datatypes(datatype_conf_path: str) -> Set[str]:
+def _parse_datatypes(datatype_conf_path) -> Set[str]:
```

Comment:
@bernt-matthias Unfortunately dropping the type annotation here hides issues when `datatype_conf_path` is processed by `galaxy.util.parse_xml()` . Do you mind if I open a separate PR that addresses them? I went down a rabbit hole for the last couple of days unfortunately :D

Reply:
> Do you mind if I open a separate PR that addresses them?

No at all. Please do so. 

> I went down a rabbit hole for the last couple of days unfortunately

Good that you found out again.